### PR TITLE
Merge 2.4 into master

### PR DESCRIPTION
**Pull request by alanpoulain**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/GraphQl/Resolver/Stage/DeserializeStage.php [hunk 1]
```diff
@@ -54,6 +54,12 @@ public function __invoke($objectToPopulate, string $resourceClass, string $opera
             $denormalizationContext[AbstractNormalizer::OBJECT_TO_POPULATE] = $objectToPopulate;
         }
 
-        return $this->denormalizer->denormalize($context['args']['input'], $resourceClass, ItemNormalizer::FORMAT, $denormalizationContext);
+        $item = $this->denormalizer->denormalize($context['args']['input'], $resourceClass, ItemNormalizer::FORMAT, $denormalizationContext);
+
```
Comment (teohhanhui) [inline src/GraphQl/Resolver/Stage/DeserializeStage.php]: Remove blank line?

Comment (alanpoulain) [reply src/GraphQl/Resolver/Stage/DeserializeStage.php]: A matter of taste, doesn't it?

Comment (teohhanhui) [reply src/GraphQl/Resolver/Stage/DeserializeStage.php]: True... Haha. But looking elsewhere in our code, this seems to be the style.

**Comment by teohhanhui** [comment]
https://github.com/symfony/symfony/issues/33450#issuecomment-527594796

Let's fix it here then.

File: src/Bridge/Symfony/Bundle/Test/Client.php [hunk 2]
```diff
@@ -212,4 +212,24 @@ public function enableReboot(): void
...
+     * @return array<string, string[]>
+     */
+    private static function extractHeaders(array $options): array
+    {
+        $headers = $options['normalized_headers'] ?? $options['headers'];
+        foreach ($headers as $key => &$values) {
+            foreach ($values as &$value) {
+                $prefix = $key.':';
+                if (0 === stripos($value, $prefix)) {
+                    $value = ltrim(substr($value, \strlen($prefix)));
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/Test/Client.php]: Hmm... This piece of code seems unnecessarily complex and error prone to me...

Comment (bpolaszek) [reply src/Bridge/Symfony/Bundle/Test/Client.php]: Passing by reference == matter of taste, but I don't see any pb in removing them.
The goal is to remove the prefix from the value: prefix is key (original-cased), clon and space. Which approach do you recommend?

File: src/Bridge/Symfony/Bundle/Test/Client.php [hunk 3]
```diff
@@ -212,4 +212,25 @@ public function enableReboot(): void
     {
         $this->kernelBrowser->enableReboot();
     }
+
+    /**
+     * Extract headers depending on the symfony/http-client version being used.
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/Test/Client.php]:
```suggestion
     * Extracts headers depending on the symfony/http-client version being used.
```

File: src/Bridge/Symfony/Bundle/Test/Client.php [hunk 4]
```diff
@@ -212,4 +212,25 @@ public function enableReboot(): void
...
+     */
+    private static function extractHeaders(array $options): array
+    {
+        if (!isset($options['normalized_headers'])) {
+            return $options['headers'];
+        }
+
+        $headers = [];
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/Test/Client.php]:
```suggestion
        /** @var array<string, string[]> $headers */
        $headers = [];
```

Looks like this should make phpstan happy. See https://github.com/phpstan/phpstan/issues/2404

**Comment by teohhanhui** [comment]
Thanks @alanpoulain, @bpolaszek ! :tada:

**Pull request merged**